samples/pollDiscoveryOccurrenceFinished.js

```diff
@@ -0,0 +1,54 @@
+// [START containeranalysis_poll_discovery_occurrence_finished]
+// Repeatedly query the Container Analysis API for the latest discovery occurrence until it is
+// either in a terminal state, or the timeout value has been exceeded
+const pollDiscoveryOccurrenceFinished = async (
+    projectId = 'your-project-id', // Your GCP Project ID
+    imageUrl = 'my-url.io/project/image', // Image to attach metadata to
+    timeoutSeconds = 'timeout-in-seconds', // The number of seconds to listen for the new Pub/Sub messages
+) => {
+    // Import the library and create a client
+    const grafeas = require('@google-cloud/grafeas');
+    const client = new grafeas.v1.GrafeasClient();
+
+    const formattedParent = client.projectPath(projectId);
+
+    let filter = `resourceUrl=\"${imageUrl}\" AND noteProjectId=\"goog-analysis\" AND noteId=\"PACKAGE_VULNERABILITY\"`;
+    // [END containeranalysis_poll_discovery_occurrence_finished]
+    filter = `kind = \"DISCOVERY\" AND resourceUrl = \"${imageUrl}\"`;
+    // [START containeranalysis_poll_discovery_occurrence_finished]
+
+    const pollDiscoveryOccurrence = () => {
+        return new Promise(async(resolve, reject) => {
+            setTimeout(() =>{
+                reject('Timeout while retrieving discovery occurrence.');
+            }, timeoutSeconds * 1000);
+            let found = false;
+            while (found == false) {
+                let [discoveryOccurrences] = await client.listOccurrences({
+                    parent: formattedParent,
+                    filter: filter
+                })
+                if (discoveryOccurrences.length > 0) {
+                    found = true;
+                    clearTimeout();
+                    resolve(discoveryOccurrences);
+                }
+            }
+        });
+    }
```

Comment:
it looks like you're missing a section here. First you need to poll for the discovery occurrence. Once you find it, you need to poll again until it's in a terminal state 

See line 261 of Ruby: https://github.com/GoogleCloudPlatform/ruby-docs-samples/pull/418/files#diff-ca8d9ed526732e2a7cf5942495e0557fR261

Reply:
currently my sample fails because the analysis status isn't getting saved with the occurrence )-: 
for example the following results in an occurrence with just `discovered: { discovered: null }`
```
    const occurrenceRequest = {
      parent: formattedParent,
      occurrence: {
        noteName: `${formattedNoteName}-discovery-polling`,
        discovered: {
          discovered: {
            analysisStatus: 'FINISHED_SUCCESS',
          },
        },
      },
      resource: {
        uri: resourceUrl,
      },
    };
```